app/src/main/java/com/google/android/soundchecker/AudioEncoderDecoderActivity.kt

```diff
@@ -851,6 +912,32 @@ class AudioEncoderDecoderActivity : ComponentActivity() {
             for (channel in 0 until numberOfChannels) {
                 mWaveforms!!.add(results[channel].buffer)
             }
+            if (mInputFile == null) {
+                mTopBins = IntArray(results.size)
+                mTopFrequencies = FloatArray(results.size)
+                if (mFundamentalFrequencies == null) {
+                    mFundamentalFrequencies = mutableListOf<MutableList<Float>?>()
+                    for (channel in 0 until results.size) {
+                        mFundamentalFrequencies!!.add(mutableListOf<Float>())
+                    }
+                }
+                for (channel in 0 until results.size) {
+                    val bins = results[channel].bins!!
+                    var maxValue = bins[0]
```

Comment:
How about using max function then the bin position can be known by indexOf? https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/max.html#

Reply:
I prefer leaving this as is as it's pretty standard. Max before indexOf has potential issue with duplicates and is harder to read in my opinion

Reply:
The existing code doesn't handle the duplicates either. Is the last maximum preferred? `Indexof` is not a rare API.

Reply:
Thanks! So much better